#60 Update release notes v1.1.0 (#74)
Inbibo/adonisfx_docs: docs/release_notes.md

- Attachment to geometry constraints in AdnMuscle and AdnRibbonMuscle solvers. Multiple target geometries are supported.
- Slide on geometry constraints in AdnMuscle and AdnRibbonMuscle solvers. Multiple target geometries are supported.

> - Shape preservation constraints integrated into all AdonisFX solvers.

Reviewer: For consistency with the previous lines

```suggestion
- Shape preservation constraints integrated in all solvers.
```

Author: Done.

- Point mass support integrated in all solvers to control the dynamics of the system.

> - Calculate the mass distribution using a physically based estimate of the volume.

Reviewer: ```suggestion
- Compute point mass distribution based on the density and volume or area.
```

Author: Modified.

- Attachment to geometry constraints in AdnMuscle and AdnRibbonMuscle solvers. Multiple target geometries are supported.
- Slide on geometry constraints in AdnMuscle and AdnRibbonMuscle solvers. Multiple target geometries are supported.

> - Shape preservation constraints integrated in all solvers.

Reviewer 2: I would specify here that Shape Preservation constraints have been created for this release and integrated them in all solvers. Like it is it sounds like we already had them in some solvers and just have integrated them in all of them.

Reviewer 3: I would list the solver names in this case

Author: Specified the solver names.

Reviewer 2: Please correct the typo, that "in" is repeated twice.

Author: Fixed.

> - Point mass support integrated in all solvers to control the dynamics of the system.

Reviewer 3: Can you please also be verbose with the deformers here? Thanks!

Author: Done.

> - Compute point mass distribution based on the density and volume or area.

Reviewer 3: I would not add the specifier about volume or area. No need to expose the internals
```suggestion
- Compute point mass distribution based on the density.
```

### Maya

- Added AdnSkinMerge deformer to blend the results from animated and simulated input geometries.
- Added support for attachment to geometry in AdnMuscle and AdnRibbonMuscle solvers.
- Added support to slide on geometry in AdnMuscle and AdnRibbonMuscle solvers.

> - Added support for shape preservation in all AdonisFX solvers.

Reviewer 4: What is the difference between this line and the one above (line 9)? If it is about debugging, maybe is worth moving it to the line about it.

Author: Added shape preservation as a new debuggable feature in line 23.

Reviewer 3: Here we are not referring to the constraints but rather the system as a whole. I would not say "all solvers" as in the future we will maybe have more deformers/solvers and it might be difficult to track it back. Can we be more verbose? List the deformers please

- Fixed import of the constraint weight arrays on sparse arrays. *AdonisFX-1155*
- Fixed a bug that caused geometry sliding constraints to not find valid triangles to slide on initialisation. *AdonisFX-1165*
- Fixed paint flood operation not normalising the weights properly. *AdonisFX-1146*
- Fixed a bug that occurred when data from the target mesh was missing. *AdonisFX-1183*

> - Fixed a bug that allowed to remove target geometries while painting Slide On Geometry constraints. *AdonisFX-1192*

Reviewer: Please remove this one

Author: It's already removed.

- Fixed a bug that allowed to open the AdonisFX Paint Tool for the final mesh of an AdnSkinMerge deformer. *AdonisFX-1198*

> - Fixed AdonisFX Paint Tool initialization of simulated nodes with intermediate nodes. *AdonisFX-1184*

Reviewer: This is ok! Because the bug was there in v1.0.0 for sure!

### Improvements

> - Allow source nodes of any type in the *Sensors Connection Editor*. *AdonisFX-1189*

Reviewer: Move this one to the list of "Maya". We probably can remove "Improvements" if there are no other items

Author: Moved. There is still one entry in improvements: "- Prevent the use of simulated muscles as geometry targets. *AdonisFX-1200*". But we might want to exposit in other way.

> - Prevent the use of simulated muscles as geometry targets. *AdonisFX-1200*

Reviewer: Please remove this one, it is not an improvement compared to the previous version

Author: Removed.

### Known Limitations

> - The *Max Sliding Distance* parameter in AdnSkin and AdnSimshape is represented in scene units. The higher this value is, the more units in the space the sliding constraint allows a vertex to slide on. If the polygons of the target geometry to slide on are very little compared to that value, then the sliding constraint will be time and memory consuming. *AdonisFX-997*

Reviewer: 997 is done. Please remove this item

Author: Removed.

Reviewer 4: Is 997 done? We added a dialog but the issue is still there. I remember that we already discuss about having to expose this one.

Author: It is true that even though we have integrated the warning dialog, if the user continues with a value that is too high for a mesh, Maya will freeze due to the initialization of the slide constraints. However, I don't know if we should expose this as a limitation since this is how slide constraints work.

That said, should we expose it as a limitation or somewhere else?

Reviewer 3: I don't think we should expose that as a limitatation. If we use for example more subdivisions for a mesh Maya will warn the user. However, if skipped, depending on the case, maya will still freeze. I think with exposing the warning we are informing the user correctly and don't need to expose it as a limitation

- Fixed a bug that caused geometry sliding constraints to not find valid triangles to slide on initialisation. *AdonisFX-1165*
- Fixed paint flood operation not normalising the weights properly. *AdonisFX-1146*

> ### Known Limitations

Reviewer 4: There is a typo at 

```Added AdnSkinMerge deformer to blend the results from animated and simualted input geometries.```

Author: Fixed.

> ### Known Limitations

Reviewer 4: Should we also include at "Bug Fixes" #1198? Despite we initialize created for AdnSkinMerge we applied to all deformers.

Author: It's already added, we would have to generalize it for all deformers.

> - The *Max Sliding Distance* parameter in AdnSkin and AdnSimshape is represented in scene units. The higher this value is, the more units in the space the sliding constraint allows a vertex to slide on. If the polygons of the target geometry to slide on are very little compared to that value, then the sliding constraint will be time and memory consuming. *AdonisFX-997*

Reviewer 4: There is no issue yet but should we mention that we do not support simulated muscles as geometry targets? @carlosmonteagudoinbibo 

We already have a note at Targets from muscle.md and ribbon.md

Author: I've already added as an improvement.

> - The activations debugger in AdnSimshape is limited to the Parallel and Serial evaluation modes. *AdonisFX-1026*

Reviewer 4: Is it not #1026 solved?

Reviewer 4: I guess that we will use that issue number. Let me resolve this conversation in order to approve the PR.

- Fixed import of the constraint weight arrays on sparse arrays. *AdonisFX-1155*
- Fixed a bug that caused geometry sliding constraints to not find valid triangles to slide on initialisation. *AdonisFX-1165*
- Fixed paint flood operation not normalising the weights properly. *AdonisFX-1146*

> - Fixed a bug that occurred when data from the target mesh was missing. *AdonisFX-1183*

Reviewer 4: `` - Fixed a bug that happened when reference mesh data was missing at AdnSkin deformer. *AdonisFX-1183```

Author: Done.

Reviewer: Please, remove this one. We don't have to list fixes for bugs that where never exposed to the user.

- Fixed a bug that allowed to remove target geometries while painting Slide On Geometry constraints. *AdonisFX-1192*

> - Fixed a bug that allowed to open the AdonisFX Paint Tool for the final mesh of an AdnSkinMerge deformer. *AdonisFX-1198*

Reviewer 4: Remove the one about #1192. If slide on geometry was not out yet, let's not say we messed up during development :)).

About #1198 I've added other comment. Let's generalize for all deformers.

Author: Right.

Reviewer: Please remove this one (same reason)

Author: Removed.

- Fixed import of the constraint weight arrays on sparse arrays. *AdonisFX-1155*
- Fixed a bug that caused geometry sliding constraints to not find valid triangles to slide on initialisation. *AdonisFX-1165*
- Fixed paint flood operation not normalising the weights properly. *AdonisFX-1146*
- Fixed AdonisFX Paint Tool initialization of simulated nodes with intermediate nodes. *AdonisFX-1184*

### Known Limitations

> - The activations debugger in AdnSimshape is limited to the Parallel and Serial evaluation modes. *AdonisFX-1026*

Reviewer 3: Leaving a comment here just in case we have more limitations to add. I'm thinking

Reviewer 3: Should we add the limitation abou the volume preservation issue?

Reviewer 3: We also encountered that the system might become unstable when we have a very high mass value in DampVelocities

Reviewer 3: I know this might not be the right way of exposing the limitation but should we mention something about the stiffness ranges?

Reviewer 3: Also is it considered a limitation to not be able to have separate stiffness per constraint?

Reviewer: > Should we add the limitation about the volume preservation issue?

Added

Reviewer 3: Resolving this comment!
